json-rpc-lambda-client/src/main/java/com/amazonaws/blox/clients/JsonRpcLambdaClient.java

```diff
@@ -0,0 +1,193 @@
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.awssdk.services.lambda.LambdaAsyncClient;
+import software.amazon.awssdk.services.lambda.model.InvokeRequest;
+import software.amazon.awssdk.services.lambda.model.InvokeResponse;
+
+/**
+ * A JSON-RPC client that translates calls into Lambda function invocations.
+ *
+ * <p>This class should not be used directly under normal circumstances. instead, use the {@link
+ * JsonRpcLambdaClient#newProxy(ObjectMapper, Class, LambdaAsyncClient, String) newProxy} factory
+ * method to create a dynamic client for a given interface.
+ */
+@Slf4j
+public class JsonRpcLambdaClient implements IJsonRpcClient {
+
+  private final LambdaAsyncClient lambda;
+  private final String functionName;
+  private final JsonRpcClient client;
+
+  private JsonRpcLambdaClient(LambdaAsyncClient lambda, String functionName) {
+    this(defaultObjectMapper(), lambda, functionName);
+  }
+
+  private JsonRpcLambdaClient(ObjectMapper mapper, LambdaAsyncClient lambda, String functionName) {
+    this.lambda = lambda;
+    this.functionName = functionName;
+    this.client = new JsonRpcClient(mapper);
+  }
+
+  private static ObjectMapper defaultObjectMapper() {
+    return new ObjectMapper();
+  }
+
+  /**
+   * Create a new dynamic JSON-RPC proxy for an interface that dispatches calls as Lambda function
+   * invocations.
+   *
+   * @param mapper the Jackson mapper to use for serializing/deserializing requests/responses
+   * @param serviceInterface the interface of the service to call
+   * @param lambdaClient a lambda client that can call the function that's running the service
+   *     implementation
+   * @param functionName the name of the labmda function that's running the service implementation
+   * @param <T>
+   * @return
+   */
+  public static <T> T newProxy(
+      ObjectMapper mapper,
+      Class<T> serviceInterface,
+      LambdaAsyncClient lambdaClient,
+      String functionName) {
+    IJsonRpcClient client = new JsonRpcLambdaClient(mapper, lambdaClient, functionName);
+    return ProxyUtil.createClientProxy(
+        Thread.currentThread().getContextClassLoader(), serviceInterface, client);
+  }
+
+  /**
+   * Create a new dynamic JSON-RPC proxy for an interface that dispatches calls as Lambda function
+   * invocations with a default Jackson ObjectMapper for serializing/deserializing payloads.
+   *
+   * @param serviceInterface the interface of the service to call
+   * @param lambdaClient a lambda client that can call the function that's running the service
+   *     implementation
+   * @param functionName the name of the labmda function that's running the service implementation
+   * @param <T>
+   * @return
+   */
+  public static <T> T newProxy(
+      Class<T> serviceInterface, LambdaAsyncClient lambdaClient, String functionName) {
+    return newProxy(defaultObjectMapper(), serviceInterface, lambdaClient, functionName);
+  }
+
+  /** {@inheritDoc} */
+  @Override
```

Comment:
what's this overriding? (and the rest) we're just implementing an interface no?

Reply:
As discussed, these are unfortunately required generic wrappers from the `IJsonRpcClient`interface for the corresponding methods that return `Object`. I'll replace the inheritdoc comment with an explanation.